### PR TITLE
Username and password length constraints

### PR DESCRIPTION
**Pull request by Jie-F**
Enforce a non-empty username and a password at least 6 characters long. Both fields now have a max of 100 characters.

### CODE REVIEW

File: code/HabitTracker/app/src/main/java/com/example/habittracker/activities/LoginActivity.java [hunk 1]
```diff
@@ -207,6 +212,20 @@ public void registerPressed(View view) throws InvalidKeySpecException, NoSuchAlg
...
+        if (username.length() == 0) {
+            Toast.makeText(LoginActivity.this, "Username cannot be empty!", Toast.LENGTH_SHORT).show();
+            return;
+        }
+        if (passwordField.getText().toString().length() < 6) {
+            Toast.makeText(LoginActivity.this, "Password must be at least 6 characters long!", Toast.LENGTH_SHORT).show();
+            return;
+        }
+        if (username.equals(passwordField.getText().toString())) {
+            Toast.makeText(LoginActivity.this, "Username cannot be identical to the password!", Toast.LENGTH_SHORT).show();
```
Comment (zarifmahfuz) [inline code/HabitTracker/app/src/main/java/com/example/habittracker/activities/LoginActivity.java:225]: What's wrong with username being identical to the password?

**Review by zarifmahfuz — COMMENTED** [review]
Looks good. Just have one question attached with this review.

File: code/HabitTracker/app/src/main/java/com/example/habittracker/activities/LoginActivity.java (hunk 1 above)
Comment (Jie-F) [reply code/HabitTracker/app/src/main/java/com/example/habittracker/activities/LoginActivity.java:225]: It is a very simple security check to make sure the password isn't super weak. We could go much further and make sure the username isn't a substring in the password, that the password isn't among a list of popular passwords, that the password contains a variety of different characters, etc.

Comment (zarifmahfuz) [reply code/HabitTracker/app/src/main/java/com/example/habittracker/activities/LoginActivity.java:225]: Okay, cool. I'll approve the PR!

**Review by zarifmahfuz — APPROVED** [review]
Please go ahead and merge

**Pull request merged**